### DISCUSSION TITLE
Artifacts on the image extracted from gif

### DISCUSSION THREAD

**Discussion by KubaZ2** [dlemstra/Magick.NET#957] [Help] [answered]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of Magick.NET
- [x] I have searched [open](https://github.com/dlemstra/Magick.NET/issues) and [closed](https://github.com/dlemstra/Magick.NET/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
I have the gif:
![gif](https://user-images.githubusercontent.com/77853483/122071182-d5f23000-cdf6-11eb-92fa-359fdfdb5de8.gif)
I use `MagickImageCollection` to storage it.
I want to get the 5th frame of it, I make it using `imageCollection[5]`, expected result:
![image](https://user-images.githubusercontent.com/77853483/122071335-f4582b80-cdf6-11eb-8baf-be9b0ae6ef14.png)
but the Magick.net gives me this:
![image](https://user-images.githubusercontent.com/77853483/122071006-af33f980-cdf6-11eb-814d-7ff0badf2f5d.png)
(it has transparent holes)

That is strange, because if I make a gif from the `imageCollection`, it looks good (to be honest, the original gif is a gif processed by Magick.net and there was the same problem with it as this issue).

### Steps to Reproduce
```cs
using ImageMagick;
using System;
using System.Net.Http;

MagickImageCollection imageCollection = new(await new HttpClient().GetStreamAsync("https://user-images.githubusercontent.com/77853483/122071182-d5f23000-cdf6-11eb-92fa-359fdfdb5de8.gif"), MagickFormat.Gif);
imageCollection.Write($"{AppDomain.CurrentDomain.BaseDirectory}gif.gif", MagickFormat.Gif);
imageCollection[5].Write($"{AppDomain.CurrentDomain.BaseDirectory}gif.png", MagickFormat.Png);
```

### System Configuration
- Magick.NET version: Magick.NET-Q8-x64 7.24.1
- Environment (Operating system, version and so on): Windows 10 Home 64-bit (10.0, Build 19042) (19041.vb_release.191206-1406)
- Additional information:
CPU: Intel Core i5-4440 3.10GHz
GPU: GeForce GTX 1060 6GB
GPU Drivers: 466.77

**Comment by dlemstra** [comment, marked as answer]
This is happening because each frame of your gif contains only the changes compared to the previous frame. You will need to call `imageCollection.Coalesce` before writing the image to make sure each frame contains all the pixels.

p.s. You should also need to do `using MagickImageCollection imageCollection`
